android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -701,11 +707,37 @@ public static void switchToNextKeyboard(Context context) {
     if (InAppKeyboard != null) {
       InAppKeyboard.setKeyboard(pkgId, kbId, langId, kbName, langName, kFont, kOskFont);
     }
+
     if (SystemKeyboard != null) {
       SystemKeyboard.setKeyboard(pkgId, kbId, langId, kbName, langName, kFont, kOskFont);
     }
   }
 
+  protected static IBinder getToken() {
+    if (IMService == null) {
+      return null;
+    }
+    final Dialog dialog = IMService.getWindow();
+    if (dialog == null) {
+      return null;
+    }
+    final Window window = dialog.getWindow();
+    if (window == null) {
+      return null;
+    }
+    return window.getAttributes().token;
+  }
+
+  public static void advanceToNextInputMode() {
+    InputMethodManager imm = (InputMethodManager) appContext.getSystemService(Context.INPUT_METHOD_SERVICE);
+
+    // TODO: this method is added in API level 16 and deprecated in API level 28
+    // Reference: https://developer.android.com/reference/android/view/inputmethod/InputMethodManager.html#switchToNextInputMethod(android.os.IBinder,%20boolean)
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
```

Comment:
Is this a static comparison based on the SDK version and not the device version? If so, it's not going to do much :)

Reply:
It's a check for the device version. Prior art in `MainActivity.java`

Reply:
I see. My bad (and bad Android SDK developers for making it look like a `CONSTANT`!)